src/Eloquent/Builder.php

```diff
@@ -203,55 +199,42 @@ public function raw($value = null)
     }
 
     /**
-     * Attempt to create the record if it does not exist with the matching attributes.
-     * If the record exists, it will be returned.
+     * Get the first record matching the attributes. If the record is not found, create it.
      *
-     * @param  array $attributes The attributes to check for duplicate records
-     * @param  array $values     The attributes to insert if no matching record is found
+     * @param  array $attributes
+     * @param  array $values
      */
-    public function createOrFirst(array $attributes = [], array $values = []): Model
+    public function firstOrCreate(array $attributes = [], array $values = []): Model|Builder
```

Comment:
When would this return a Builder instance instead of a Model?

Reply:
Never, I was fooled by the `@return Model|static` [in the parent class](https://github.com/laravel/framework/blob/97238d27e271d6f8b36b64680756f788130f19c5/src/Illuminate/Database/Eloquent/Builder.php#L571).